src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -209,13 +204,26 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                     request.Headers.TransferEncodingChunked = true;
                 }
 
+                if (request.Version.Major != 1)
+                {
+                    // TODO #23134: Support 2.0
+                    throw new PlatformNotSupportedException($"Only HTTP 1.0 supported -- request.Version was {request.Version}");
+                }
+                else if (request.Version.Minor == 0 && request.Headers.TransferEncodingChunked == true)
+                {
+                    // HTTP 1.0 does not support chunking
+                    throw new NotSupportedException($"HTTP 1.0 does not support chunking.");
+                }
```

Comment:
How do the existing handlers (WinHttpHandler, CurlHandler, HttpClientHandler on desktop, etc.) behave when a 1.x version is supplied with an x other than 0 or 1?  I assume they don't fail?

Reply:
.NET Framework will throw exception if passed a version field other than (1,1) or (1.0).

WinHttpHandler will use 1.1 for any unrecognized HttpRequestMessage.Version field.